scripts/build_demand.py

```diff
@@ -0,0 +1,176 @@
+
+    Parameters
+    ----------
+    n : pypsa network
+    regions : .geojson
+        Contains bus_id of low voltage substations and
+        bus region shapes (voronoi cells)
+    load_paths: paths of the load files
+    admin_shapes : .geojson
+        contains subregional gdp, population and shape data
+    countries : list
+        List of countries that is config input
+    scale : float
+        The scale factor is multiplied with the load (1.3 = 30% more load)
+
+    Returns
+    -------
+    n : pypsa network
+        Now attached with load time series
+    """
+    substation_lv_i = n.buses.index[n.buses["substation_lv"]]
+    regions = gpd.read_file(regions).set_index("name").reindex(substation_lv_i)
+
+    load_paths = load_paths
+    # Merge load .nc files: https://stackoverflow.com/questions/47226429/join-merge-multiple-netcdf-files-using-xarray
+    gegis_load = xr.open_mfdataset(load_paths, combine="nested")
+    gegis_load = gegis_load.to_dataframe().reset_index().set_index("time")
+    # filter load for analysed countries
+    gegis_load = gegis_load.loc[gegis_load.region_code.isin(countries)]
+    logger.info(f"Load data scaled with scalling factor {scale}.")
+    gegis_load["Electricity demand"] *= scale
+    shapes = gpd.read_file(admin_shapes).set_index("GADM_ID")
+    shapes["geometry"] = shapes["geometry"].apply(lambda x: make_valid(x))
+
+    def upsample(cntry, group):
+        """
+        Distributes load in country according to population and gdp
+        """
+        l = gegis_load.loc[gegis_load.region_code == cntry]["Electricity demand"]
+        if len(group) == 1:
+            return pd.DataFrame({group.index[0]: l})
+        else:
+            shapes_cntry = shapes.loc[shapes.country == cntry]
+            transfer = vtransfer.Shapes2Shapes(
+                group, shapes_cntry.geometry, normed=False
+            ).T.tocsr()
+            gdp_n = pd.Series(
+                transfer.dot(shapes_cntry["gdp"].fillna(1.0).values), index=group.index
+            )
+            pop_n = pd.Series(
+                transfer.dot(shapes_cntry["pop"].fillna(1.0).values), index=group.index
+            )
+
+            # relative factors 0.6 and 0.4 have been determined from a linear
+            # regression on the country to EU continent load data
+            # (refer to vresutils.load._upsampling_weights)
+            # TODO: require adjustment for Africa
+            factors = normed(0.6 * normed(gdp_n) + 0.4 * normed(pop_n))
+            return pd.DataFrame(
+                factors.values * l.values[:, np.newaxis],
+                index=l.index,
+                columns=factors.index,
+            )
+
+    load = pd.concat(
+        [
+            upsample(cntry, group)
+            for cntry, group in regions.geometry.groupby(regions.country)
+        ],
+        axis=1,
+    )
+
+    n.madd("Load", substation_lv_i, bus=substation_lv_i, p_set=load)
```

Comment:
This was done partially, meaning that the attach_load function now is back in add_electricity and kind of works, it just needs some adjustments, which I will do tomorrow :)

Reply:
Can we not get rid of the `substation_lv_i` stuff in `add_electricity.py`?
It would be great if all that stuff is dealt with in `build_demand_profiles.py` and that the `demand_profiles.csv` contains all the information such that `substation_lv_i` is not needed.